include/kelcoro/thread_pool.hpp

```diff
@@ -59,6 +62,28 @@ struct task_queue {
     }
   }
 
+  void push_list(task_node* node) {
```

Comment:
i think its better to provide overloading push(node* first, node* last) and reuse it both in push_list and push to not repeat logic with notfy / lock